### ISSUE TITLE
[BUG] SurveyMonkey - Survey Response not providing additional details

### ISSUE THREAD

**Issue by ctrlaltdylan** [PipedreamHQ/pipedream#4373]
**Describe the bug**
We are unable to retrieve the full details of a survey answer, only the basic details of the survey response are included.

It's possible this  `this.getSummary()` method in the `sources/common/base.mjs` is the culprit:

https://github.com/PipedreamHQ/pipedream/blob/4668af2bb91f1b7bc4d910e9bbb28d392d862b22/components/survey_monkey/sources/common/base.mjs#L27

**To Reproduce**
1. Create a new workflow
2. Configure a new Survey Monkey - New Survey Response trigger
3. Enter a survey
4. Only see basic details about the survey response in the emitted event data

**Expected behavior**
I expect to see the entire survey monkey response results in the emitted event.

**Additional context**
Reported by ActForPeace in a private Slack channel.

**Comment by tsadler4857** [comment]
Some further information on the issue we have seen...

The initial event received includes limited information with only collector_id, recipient_id, respondent_id, survey_id and user_id.  These values are enough to make a follow up call to SurveyMonkey to get the response with its answers, however the pipedream integration doesn't allow for the scope "response_details" and so we are unable to get the answers to the questions in the response.

We are working around this with our own temporary app in SurveyMonkey, and we are retrieving the answers using the endpoint:

https://api.surveymonkey.com/v3/surveys/${this.surveyid}/responses/${this.responseid}/details

I suspect the issue is in part in the configuration of the Pipedream app in SurveyMonkey?

**Comment by dannyroosevelt** [comment]
@tsadler4857 we have `response_details` as an optional scope on the Pipedream OAuth app for SurveyMonkey, because that requires a paid plan on SurveyMonkey's end.

![Screenshot by Dropbox Capture](https://user-images.githubusercontent.com/20780753/193187909-cac15e3b-1423-49dd-a381-79fc34db1bed.png)

Can you clarify what you mean re: your own temporary app in SurveyMonkey? Is this a custom OAuth app you created? Or something else?

**Comment by tsadler4857** [comment]
Hi,

First, the temporary app is that we have created a private app in draft mode with the "View Response Detalils" scope as required.  This doesn't require us to have the relevant team plan until that draft expires - which is why it is temporary.

We have also created a public app with the same scope required, but we aren't keen to publish a public app and I'm not sure that SurveyMonkey would let us!

I have connected to two SurveyMonkey accounts using the pipedream built-in option, but wasn't presented with the option to include View Response Details scope.  both these accounts are paid accounts with SurveyMonkey.

Perhaps the issue for us is just that we can't find out how to select that additional scope when setting up the connection?

**Comment by tsadler4857** [comment]
A quick update from my end.  I have spoken too soon, but think something may have changed... I'm not quite sure.  My colleague was setting some of this up, but I also worked on it, but I just reconnected using the pipedream surveymonkey app and could now see how to include that response_details scope, which I have done.

I will do some testing and make sure it does what we need. But this might be all good!

Sorry, for that confusion!  I'll confirm here that all is well when I've been able to check it.

(I must be going mad to have not seen that option in the connection... I promise I looked for it!!!)
